### ISSUE TITLE
Possible mistake in consistency loss when applying mask

### ISSUE THREAD

**Issue by barskern** [microsoft/Semi-supervised-learning#96]
https://github.com/microsoft/Semi-supervised-learning/blob/5c9ee2148ab1a30e63fa05414f12033c6952d1fc/semilearn/core/criterions/consistency.py#L33-L37

When applying a mask, some entries in the `loss` tensor, are **zeroed** out, then the mean of the tensor is taken. The way I read it, the mean will now include the entries which have been zeroed out, hence the mean will account for the entries which are masked out. Is this interpretation correct?

The way I think about it, it should rather be:

```python
if mask is not None:
    loss = loss * mask
    return loss.sum() / torch.maximum(mask.sum(), torch.tensor(1.0))
else:
    return loss.mean()
```

This will find the mean of the nonzero entries of `loss`.

**Comment by Hhhhhhao** [comment]
Hi, we inherit this implementation from TorchSSL. This should not cast too large difference in results since the actual difference only lies in the scale of the gradient. 

Let me know if you find the modified version has better performance, in that case we can move this part of code into the codebase.
